src/transformers/models/llava/processing_llava.py

```diff
@@ -0,0 +1,220 @@
+        vision_model (`CLIPVisionModel`):
+            An instance of ['CLIPVisionModel`]. The Vision Model is a required input.
+    """
+    attributes = ["image_processor", "tokenizer"]
+    tokenizer_class = "AutoTokenizer"
+    image_processor_class = "CLIPImageProcessor"
+
+    def __init__(self, image_processor, tokenizer, vision_model):
+        super().__init__(image_processor, tokenizer)
+
+        self.vision_model = vision_model
+        self.DEFAULT_IMAGE_TOKEN = "<image>"
+        self.IMAGE_TOKEN_INDEX = -200
+
+    def __call__(
+        self,
+        images: ImageInput = None,
+        text: Union[TextInput, List[TextInput]] = None,
+        add_special_tokens: bool = True,
+        padding: Union[bool, str, PaddingStrategy] = False,
+        truncation: Union[bool, str, TruncationStrategy] = None,
+        max_length: Optional[int] = None,
+        stride: int = 0,
+        pad_to_multiple_of: Optional[int] = None,
+        return_attention_mask: Optional[bool] = None,
+        return_overflowing_tokens: bool = False,
+        return_special_tokens_mask: bool = False,
+        return_offsets_mapping: bool = False,
+        return_token_type_ids: bool = False,
+        return_length: bool = False,
+        verbose: bool = True,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        This method uses [`CLIPProcessor.__call__`] method to prepare image(s) for the model, and
+        [`LlamaTokenizer.__call__`] to prepare text for the model.
+
+        Please refer to the docstring of the above two methods for more information.
+        """
+
+        if images is None and text is None:
+            raise ValueError("You have to specify at least images or text.")
+
+        encoding = BatchFeature()
+        dummy = {}
+        if text is not None:
+            text = self.DEFAULT_IMAGE_TOKEN + "\n" + text + "###"
+            prompt_chunks = [
+                self.tokenizer(
+                    chunk,
+                    add_special_tokens=add_special_tokens,
+                    padding=padding,
+                    truncation=truncation,
+                    max_length=max_length,
+                    stride=stride,
+                    pad_to_multiple_of=pad_to_multiple_of,
+                    return_attention_mask=return_attention_mask,
+                    return_overflowing_tokens=return_overflowing_tokens,
+                    return_special_tokens_mask=return_special_tokens_mask,
+                    return_offsets_mapping=return_offsets_mapping,
+                    return_token_type_ids=return_token_type_ids,
+                    return_length=return_length,
+                    verbose=verbose,
+                    # return_tensors=return_tensors,
+                ).input_ids
+                for chunk in text.split("<image>")
+            ]
+
+            def insert_separator(X, sep):
```

Comment:
Could you consider renaming `X` to a more descriptive variable name? Maybe `word`, `token`?